source/server/data_moniker_service.cpp

```diff
@@ -103,20 +111,31 @@ void DataMonikerService::InitiateMonikerList(const MonikerList& monikers, Endpoi
   }
 }
 
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+#ifndef _WIN32
+void set_moniker_stream_processor(int stream_processor)
+{
+  if(stream_processor >= 0) {
+    cpu_set_t cpuSet;
+    CPU_ZERO(&cpuSet);
+    CPU_SET(stream_processor, &cpuSet);
+    sched_setaffinity(0, sizeof(cpu_set_t), &cpuSet);
+  }
+}
+#endif
+
 //---------------------------------------------------------------------
 //---------------------------------------------------------------------
 void DataMonikerService::RunSidebandReadWriteLoop(string sidebandIdentifier, ::SidebandStrategy strategy, EndpointList* readers, EndpointList* writers)
 {
 #ifndef _WIN32
-  if (strategy == ::SidebandStrategy::RDMA_LOW_LATENCY ||
-      strategy == ::SidebandStrategy::SOCKETS_LOW_LATENCY) {
+  if ((strategy == ::SidebandStrategy::RDMA_LOW_LATENCY ||
+      strategy == ::SidebandStrategy::SOCKETS_LOW_LATENCY) && c_StreamProcessor.moniker_sideband_stream_read_write >= 0) {
     pid_t threadId = syscall(SYS_gettid);
     ::SysFsWrite("/dev/cgroup/cpuset/LabVIEW_tl_set/tasks", std::to_string(threadId));
 
-    cpu_set_t cpuSet;
-    CPU_ZERO(&cpuSet);
-    CPU_SET(8, &cpuSet);
-    sched_setaffinity(threadId, sizeof(cpu_set_t), &cpuSet);
+    set_moniker_stream_processor(c_StreamProcessor.moniker_sideband_stream_read_write);
```

Comment:
This code should set the cpu affinity on a specific thread id. the setter function is setting it on thread 0 which I think is for the whole process.

Reply:
As discussed in chat, passing thread id as 0 to `sched_setaffinity `means it will set the affinity for the calling thread, which is what we need and hence we don't need to get the calling threadId explicitely and pass it here.